### ISSUE TITLE
Nim 0.12.0 supported?

### ISSUE THREAD

**Issue by oderwat** [yglukhov/coverage#1]
I have problems to compile the example in Nim 0.12.0.

``` nim
lib/pure/nativesockets.nim(15, 8) Warning: unsigned is deprecated [Deprecated]
lib/pure/net.nim(13, 37) Warning: unsigned is deprecated [Deprecated]
lib/pure/selectors.nim(12, 12) Warning: unsigned is deprecated [Deprecated]
lib/pure/selectors.nim(232, 13) Warning: mget is deprecated [Deprecated]
lib/pure/asyncnet.nim(459, 7) Warning: raiseOSError is deprecated [Deprecated]
example_1.nim(13, 43) template/generic instantiation from here
lib/system.nim(2301, 19) Error: undeclared field: 'data'
```

**Comment by yglukhov** [comment]
The `tests/test.nim` seems to work for me. Mind sharing your `example_1.nim` source?

**Comment by oderwat** [comment]
Its the code from the ReadMe

``` nim
import coverage

proc myProcToCover(x: int) {.cov.} = # Add cov pragma to proc definition to enable code coverage.
  if x == 0:
    echo "x is 0"
  else:
    echo "x is ", x

# Run your program or unittest
myProcToCover(1)

# At the end of the program, display coverage results:
echo "BY FILE: ", coveragePercentageByFile()
# Outputs: BY FILE: {test.nim: 0.5}

echo "TOTAL: ", totalCoverage()
# Outputs: TOTAL: 0.5

# Finer grained information may accessed with coverageInfoByFile proc.
```

**Comment by yglukhov** [comment]
Oops. Fixed: 0dd349458a15ad1bf7e0a7d49ed773e21953b4cb.
